Spanner/tests/Unit/Batch/BatchSnapshotTest.php

```diff
@@ -153,6 +168,15 @@ public function testPartitionQuery()
                 'maxPartitions' => $opts['maxPartitions'],
                 'partitionSizeBytes' => $opts['partitionSizeBytes']
             ])
+        ];
+
+        // This loop adds extra variable options to the arguments array.
+        foreach ($testCaseOptions as $key => $value){
+            $arguments[] = Argument::withEntry($key, $value);
+        }
+
+        $this->connection->partitionQuery(call_user_func_array(
```

Comment:
I think we can write this without `call_user_func_array` if its just for mocking.

Reply:
Oh interesting, how!? 🚀
Can you please share the idea?

Reply:
You can also do like this:
```
$expectedArgs = [
            'session' => self::SESSION,
            'partitionOptions' => [
                'maxPartitions' => $opts['maxPartitions'],
                'partitionSizeBytes' => $opts['partitionSizeBytes']
            ]
        ];
        $expectedArgs += $testCaseOptions;
        $this->connection->partitionQuery(
            Argument::that(function ($providedArgs) use ($expectedArgs) {
                return $providedArgs === $expectedArgs;
            })
        )
        ->shouldBeCalled()->willReturn(....);
```
its much shorter and neater.

Reply:
Oh wow, thanks for sharing this!!
Just a small change from that approach is that we should do a `==` instead of `===` because the later return false when the ordering is different even with all the key, values being the same.